lib/internal/main/watch_mode.js

```diff
@@ -37,10 +37,40 @@ const kWatchedPaths = ArrayPrototypeMap(getOptionValue('--watch-path'), (path) =
 const kPreserveOutput = getOptionValue('--watch-preserve-output');
 const kCommand = ArrayPrototypeSlice(process.argv, 1);
 const kCommandStr = inspect(ArrayPrototypeJoin(kCommand, ' '));
-const args = ArrayPrototypeFilter(process.execArgv, (arg, i, arr) =>
-  !StringPrototypeStartsWith(arg, '--watch-path') &&
-  (!arr[i - 1] || !StringPrototypeStartsWith(arr[i - 1], '--watch-path')) &&
-  arg !== '--watch' && !StringPrototypeStartsWith(arg, '--watch=') && arg !== '--watch-preserve-output');
+
+const watchOptions = ['--watch', '--watch-path', '--watch-preserve-output'];
+
+function argIsWatchOptionWithValue(arg) {
+  return watchOptions.some((watchOption) => StringPrototypeStartsWith(arg, `${watchOption}=`));
+}
+
+function argIsWatchOption(arg) {
+  return watchOptions.includes(arg) || argIsWatchOptionWithValue(arg);
+}
+
+const args = ArrayPrototypeFilter(process.execArgv, (arg, i, arr) => {
```

Comment:
Why not use `getOption()`? (This seems to re-invent the wheel)

Reply:
I need to filter out all watch-related options in order to pass them to the spawned option

Reply:
thank you, done